packages/meteor-wallet/src/lib/meteor-wallet.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  connect,
+  keyStores,
+  transactions as nearTransactions,
+  utils,
+} from "near-api-js";
+import {
+  InjectedWallet,
+  Network,
+  Optional,
+  Transaction,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+} from "@near-wallet-selector/core";
+import {
+  EMeteorWalletSignInType,
+  IOMeteorWalletSdk_RequestSignIn_Inputs,
+  MeteorWallet as MeteorWalletSdk,
+} from "@meteorwallet/sdk";
+import { notNullEmptyArray } from "./utils/basic_utils";
+import {
+  MeteorWalletParams_Injected,
+  MeteorWalletState,
+} from "./meteor-wallet-types";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+
+const setupWalletState = async (
+  params: MeteorWalletParams_Injected,
+  network: Network
+): Promise<MeteorWalletState> => {
+  const keyStore = new keyStores.BrowserLocalStorageKeyStore();
+
+  const near = await connect({
+    keyStore,
+    ...network,
+    headers: {},
+  });
+
+  const wallet = new MeteorWalletSdk({ near, appKeyPrefix: "near_app" });
+
+  // Cleanup up any pending keys (cancelled logins).
+  if (!wallet.isSignedIn()) {
+    await keyStore.clear();
+  }
+
+  return {
+    wallet,
+    keyStore,
+  };
+};
+
+const createMeteorWalletInjected: WalletBehaviourFactory<
+  InjectedWallet,
+  { params: MeteorWalletParams_Injected }
+> = async ({ options, logger, store, params }) => {
+  const _state = await setupWalletState(params, options.network);
+
+  const cleanup = () => {
+    _state.keyStore.clear();
+  };
+
+  const getAccounts = () => {
+    const accountId = _state.wallet.getAccountId();
+
+    if (!accountId) {
+      return [];
+    }
+
+    return [{ accountId }];
+  };
+
+  const transformTransactions = async (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    const account = _state.wallet.account()!;
+    const { networkId, signer, provider } = account.connection;
+
+    const localKey = await signer.getPublicKey(account.accountId, networkId);
+
+    return Promise.all(
+      transactions.map(async (transaction, index) => {
+        const actions = transaction.actions.map((action) =>
+          createAction(action)
+        );
+        const accessKey = await account.accessKeyForTransaction(
```

Comment:
What happens for transactions that require a `FullAccess` key?

Reply:
Yes for now we don't allow any `FullAccess` key transactions.